### ISSUE TITLE
Mailpile does not properly save metadata index on Mac OS 10.10.2.

### ISSUE THREAD

**Issue by necopinus** [mailpile/Mailpile#1234]
Steps to reproduce:

1. Start up a fresh install of Mailpile Beta II (2014-01-24 release) on Mac OS 10.10.2.
2. Add a local Maildir resource. Whether or not messages are copied to Mailpile doesn't make a difference. Whether or not you import a single Maildir, or a folder containing multiple Maildirs doesn't make a difference.
3. Complete setup, going to the inbox.
4. Quit Mailpile, either via the menu item or the command prompt. Whether or not you wait for all your mail to be imported doesn't make a difference.
5. Re-open Mailpile.
6. Mailpile prompt reports "**Corrupt data in metadata index! Trying to cope...**" After a time it *may* raise the exception "**Your metadata index is either too old, too new or corrupt!**" It may also hang indefinitely. WebUI presents the initial setup and may or may not subsequently hang if you try to re-configure Mailpile.

So far, this is reproducible for me 100% of the time. Mailpile logs do not seem to record any errors.

The net result of this is that, while Mailpile can be successfully configured and used, once shut down its configuration directory must be manually deleted and Mailpile itself set up again from scratch. :-(

**Comment by BjarniRunar** [comment]
This is bad! Thanks for reporting!

Do you know if Mailpile had finished generating a GPG key before shutting down? We recently switched to generating 4096 bit keys, which can take a significant amount of time in some cases. I think there *may* be bugs like this if the user shuts down Mailpile before the key has been created.

**Comment by necopinus** [comment]
I actually already have a 4096 bit key (I currently use Mac OS X's default Mail client + the GPGSuite).

I just tried moving my ~/.gnupg directory out of the way to force Mailpile to generate its own key, and the issue seems to be resolved. But if I revert back to using my existing keyring (and hence my current GPG key), the issue re-appears. So the issue seems to be with *existing* keys, rather than new ones generated by Mailpile.

During the course of testing this out, I ran into what seems to be a separate issue, #1238 (my key has an expiration date)... But perhaps the two are related?
